bundle/direct/dresources/adapter.go

```diff
@@ -94,6 +97,9 @@ type IResourceWithRefresh interface {
 
 	// WaitAfterUpdate waits for the resource to become ready after update.
 	WaitAfterUpdate(ctx context.Context, newState any) (newRemoteState any, e error)
+
+	// ClassifyChange classifies a set of changes using custom logic.
+	ClassifyChange(changes []structdiff.Change, newState any) (actionType deployplan.ActionType, newRemoteState any, e error)
```

Comment:
Why do we need a version of ClassifyChange that returns a new remote state?

Reply:
Yes, good point, we don't really need it